rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/robolectric/widget/OneScanCovidRDTBarcodeFactoryTest.java

```diff
@@ -0,0 +1,47 @@
+package io.ona.rdt.robolectric.widget;
+
+import com.vijay.jsonwizard.activities.JsonFormActivity;
+import com.vijay.jsonwizard.domain.WidgetArgs;
+
+import org.json.JSONException;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.powermock.reflect.Whitebox;
+
+import java.util.Date;
+
+import io.ona.rdt.application.RDTApplication;
+import io.ona.rdt.fragment.RDTJsonFormFragment;
+import io.ona.rdt.widget.OneScanCovidRDTBarcodeFactory;
+
+public class OneScanCovidRDTBarcodeFactoryTest extends WidgetFactoryRobolectricTest {
+
+    private OneScanCovidRDTBarcodeFactory oneScanCovidRDTBarcodeFactory;
+
+    private WidgetArgs widgetArgs;
+    private JsonFormActivity jsonFormActivity;
+
+    @Before
+    public void setUp() throws JSONException {
+        super.setUp();
+        oneScanCovidRDTBarcodeFactory = new OneScanCovidRDTBarcodeFactory();
+        setWidgetArgs();
+        Whitebox.setInternalState(oneScanCovidRDTBarcodeFactory, "stepStateConfig", RDTApplication.getInstance().getStepStateConfiguration());
+    }
+
+    @Test
+    public void testMoveToNextStepShouldGotoTwoSensorTriggered() throws Exception {
+        Whitebox.invokeMethod(oneScanCovidRDTBarcodeFactory, "moveToNextStep", true, Mockito.mock(Date.class));
```

Comment:
There are no assertions made here.

Could you verify that there was navigation to a different page.

Reply:
fixed.